### PR TITLE
Transform behaviour+documentation

### PR DESCRIPTION
**Pull request by djdameln**
## 📝 Description

- This PR implements a minor change to the transforms workflow. When only a `train_transform` (and not an `eval_transform`) is passed to the datamodule, Anomalib will now use the default model-specific transform as the eval transform. Previously it would re-use the `train_transform` to prevent shape mismatch. We now also raise a warning to urge the user to check their transform configs if only a `train_transform` or `eval_transform` is provided. The new behaviour may be a bit less confusing to users.
- Adds documentation for the transform functionality
- Updates the data configs, which still defined some depreceted arguments (this was a leftover from v0.x)

## Fixes
#2020 

Select what type of change your PR is:

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔨 Refactor (non-breaking change which refactors the code base)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [ ] 🔒 Security update

## ✅ Checklist

Before you submit your pull request, please make sure you have completed the following steps:

- [x] 📋 I have summarized my changes in the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) and followed the guidelines for my type of change (skip for minor changes, documentation updates, and test enhancements).
- [x] 📚 I have made the necessary updates to the documentation (if applicable).
- [ ] 🧪 I have written tests that support my changes and prove that my fix is effective or my feature works (if applicable).

For more information about code review checklists, see the [Code Review Checklist](https://github.com/openvinotoolkit/anomalib/blob/main/docs/source/markdown/guides/developer/code_review_checklist.md).

### CODE REVIEW

File: docs/source/snippets/data/transforms/datamodule_train_eval_cli.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        - class_path: torchvision.transforms.v2.RandomHorizontalFlip
+          init_args:
+            p: 0.5
+        - class_path: torchvision.transforms.v2.Resize
+          init_args:
+            size: [256, 256]
+        - class_path: torchvision.transforms.v2.Normalize
+          init_args:
+            mean: [0.485, 0.456, 0.406]
+            std: [0.229, 0.224, 0.225]
```
Comment (samet-akcay) [inline docs/source/snippets/data/transforms/datamodule_train_eval_cli.yaml:32]: I was not aware if we support this. Are we certain that this is supported? @djdameln @ashwinvaidya17 ?

Comment (djdameln) [reply docs/source/snippets/data/transforms/datamodule_train_eval_cli.yaml:32]: I tested this and it works :)

File: docs/source/snippets/data/transforms/inference_cli.sh [hunk 2]
```diff
@@ -0,0 +1,2 @@
+anomalib fit --model Patchcore --data mvtec.yaml --default_root_dit export_path
```
Comment (ashwinvaidya17) [inline docs/source/snippets/data/transforms/inference_cli.sh:1]:
```suggestion
anomalib fit --model Patchcore --data mvtec.yaml --default_root_dir export_path
```

**Review by ashwinvaidya17 — COMMENTED** [review]
Thanks for the efforts! I have one minor comment

**Review by samet-akcay — APPROVED** [review]

**Review by ashwinvaidya17 — APPROVED** [review]

**Pull request merged**